core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -641,6 +643,28 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def sendChangeNotifications(workspaceName: WorkspaceName) = {
+
+    def getUserEmails = {
+      dataSource.inTransaction{ dataAccess =>
+        withWorkspaceContext(workspaceName, dataAccess) {workspaceContext =>
+          requireAccessIgnoreLock(workspaceContext.workspace, WorkspaceAccessLevels.Write, dataAccess) {
```

Comment:
This is the same pattern as everywhere else so I'm not requesting a change here. However, it occurs to me that this authorization code is pushed further into the code than I'd like. I think it would be easier to read if it were more declarative, or at least closer to the top/left of the service function. Just planting the thought in case anyone is inspired to improve this.